homeassistant/components/roborock/__init__.py

```diff
@@ -158,14 +160,9 @@ async def setup_device(
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Handle removal of an entry."""
-    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
-    if unload_ok:
-        await asyncio.gather(
-            *(
-                coordinator.release()
-                for coordinator in hass.data[DOMAIN][entry.entry_id].values()
-            )
-        )
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
+        for coordinator in hass.data[DOMAIN][entry.entry_id].values():
+            await coordinator.release()
```

Comment:
Why aren't we using any more `gather` or TaskGroup to wait for all tasks at once?
We are now waiting one by one

Reply:
Fixed.